### PR TITLE
helm chart: initial commit for helm chart to bring up zot

### PR DESCRIPTION
**Pull request by rchincha**
Signed-off-by: Ramkumar Chinchani <rchincha@cisco.com>

**What type of PR is this?**
feature

**Which issue does this PR fix**:
issue #293 

**What does this PR do / Why do we need it**:
adds a helm chart to deploy zot registry

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: helm/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
```
Comment (adodon2go) [inline helm/templates/deployment.yaml:42]: I think we need to replace this to /v2/_catalog or something equivalent that will return 200 on both minimal & extended versions. Similar comment for readinessProbe

File: helm/templates/tests/test-connection.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ include "helm.fullname" . }}-test-connection"
+  labels:
+    {{- include "helm.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
```
Comment (adodon2go) [inline helm/templates/tests/test-connection.yaml:8]: I think this should be test-success here for v2: https://helm.sh/docs/topics/chart_tests/

File: helm/values.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# Default values for helm.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: ghcr.io/project-zot/zot
+  pullPolicy: IfNotPresent
```
Comment (adodon2go) [inline helm/values.yaml:9]: Note: on GCS currently there is no pullPolicy defined in k8s files which defaults to Never (expects image to already be there by previously loading it)

File: helm/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  echo http://$NODE_IP:$NODE_PORT
+{{- else if contains "LoadBalancer" .Values.service.type }}
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get --namespace {{ .Release.Namespace }} svc -w {{ include "helm.fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ include "helm.fullname" . }} --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+  echo http://$SERVICE_IP:{{ .Values.service.port }}
+{{- else if contains "ClusterIP" .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app.kubernetes.io/name={{ include "helm.name" . }},app.kubernetes.io/instance={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  export CONTAINER_PORT=$(kubectl get pod --namespace {{ .Release.Namespace }} $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
```
Comment (adodon2go) [inline helm/templates/NOTES.txt:20]: Should we change this to 80 as defined in helm/values.yaml by default?

**Comment by andaaron** [comment]
Will continue work in https://github.com/project-zot/zot/pull/501

**Pull request closed without merging**